### PR TITLE
disttask, ddl: move backend ctx register code to scheduler

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

Previously, `backendCtx` is registered & unregistered every time the subtasks enter the next step. This is quite unefficient because construction & destruction take times. 

The previous implementation also has the following drawbacks:
- It is easy to forget unregistering after a stage complete, causing resource leakage.
- The context that should be reused accross multiple stages maybe unexpectedly canceled.

### What is changed and how it works?

- Move register code to initialization of `Scheduler`.
- Add `Close()` method to `Scheduler`, and unregister `backendCtx` in it.
- Code refactoring.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/stage_scheduler.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ type BackfillSubTaskMeta struct {
 }
 
 // NewBackfillSchedulerHandle creates a new backfill scheduler.
-func NewBackfillSchedulerHandle(ctx context.Context, taskMeta []byte, d *ddl,
-	stage int64, summary *execute.Summary) (execute.SubtaskExecutor, error) {
+func NewBackfillSchedulerHandle(_ context.Context, taskMeta []byte, d *ddl,
```
Comment (ywqzzy) [inline ddl/stage_scheduler.go:56]:
```suggestion
func NewBackfillExecutor_ context.Context, taskMeta []byte, d *ddl,
```

**Review by ywqzzy — APPROVED** [review]
Rest LGTM

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**